src/internal/glpk_support.c

```diff
@@ -32,6 +32,35 @@
 
 #include <stdio.h>
 
+IGRAPH_THREAD_LOCAL igraph_i_glpk_error_info_t igraph_i_glpk_error_info;
+
+int igraph_i_glpk_terminal_hook(void *info, const char *s) {
+    IGRAPH_UNUSED(info);
+
+    if (igraph_i_interruption_handler) {
+        if (!igraph_i_glpk_error_info.is_interrupted && igraph_allow_interruption(NULL) != IGRAPH_SUCCESS) {
+            /* If an interruption has already occurred, do not set another error. */
+            igraph_i_glpk_error_info.is_interrupted = 1;
+            glp_error("GLPK was interrupted.");
+        } else {
+            /* Copy the error messages into a buffer for later reporting */
+            const size_t n = sizeof(igraph_i_glpk_error_info.msg) / sizeof(char) - 1;
+            while (*s != '\0' && igraph_i_glpk_error_info.msg_ptr < igraph_i_glpk_error_info.msg + n) {
```

Comment:
This is essentially an `strncpy()`, isn't it?

Reply:
Yes, it is, but it seemed that due to the need of keeping track of the end of the string, and maintaining the null terminator, `strncpy` would not have simplified the code.